pkg/elasticsearch/templates.go

```diff
@@ -67,13 +69,100 @@ func (ec *esClient) ListTemplates() (sets.String, error) {
 	return response, nil
 }
 
-func (ec *esClient) GetIndexTemplates() (map[string]interface{}, error) {
+func (ec *esClient) GetIndexTemplates() (map[string]estypes.GetIndexTemplate, error) {
 	payload := &EsRequest{
 		Method: http.MethodGet,
 		URI:    "_template/common.*",
 	}
 
 	ec.fnSendEsRequest(ec.cluster, ec.namespace, payload, ec.k8sClient)
 
-	return payload.ResponseBody, payload.Error
+	// unmarshal response body and return that
+	templates := map[string]estypes.GetIndexTemplate{}
+	err := json.Unmarshal([]byte(payload.RawResponseBody), &templates)
+	if err != nil {
+		return templates, fmt.Errorf("failed decoding raw response body into `map[string]estypes.GetIndexTemplate` for %s in namespace %s: %v", ec.cluster, ec.namespace, err)
+	}
+
+	return templates, payload.Error
+}
+
+func (ec *esClient) updateAllIndexTemplateReplicas(replicaCount int32) (bool, error) {
+
+	// get the index template and then update the replica and put it
+	indexTemplates, err := ec.GetIndexTemplates()
+	if err != nil {
+		return false, err
+	}
+
+	replicaString := fmt.Sprintf("%d", replicaCount)
+
+	for templateName, template := range indexTemplates {
+
+		currentReplicas := template.Settings.Index.NumberOfReplicas
+		if currentReplicas != replicaString {
+			template.Settings.Index.NumberOfReplicas = replicaString
+
+			templateJson, _ := json.Marshal(template)
```

Comment:
What is the reason to shallow the error here?

Reply:
copy pasta... we shouldn't run into a marshalling error since we have a struct that we're using to ensure proper type casting and structure... probably should catch the error just to be defensive